test/ring/mock/request_test.clj

```diff
@@ -131,3 +135,24 @@
       (is (instance? java.io.InputStream (:body resp)))
       (is (= (slurp (:body resp)) "foo"))
       (is (= (:content-length resp) 3)))))
+
+(defmacro when-clojure-spec
+  [& body]
+  (when (try
+          (require '[clojure.spec :as s])
+          true
+          (catch Exception _
+            (binding [*out* *err*]
+              (println "ring-mock: Skipping ring-spec tests."))
+            false))
+    `(do
+       (require 'ring.core.spec)
+       ~@body)))
+
+(deftest test-specs
+  (when-clojure-spec
+    (doseq [req [(-> (request :get "/foo/bar") (query-string nil))
+                   (request :put "/foo/bar")
+                   (request :something "/foo" {:params true})]]
```

Comment:
Alignment Is off here. Clojure mode is really struggling with the indentation around `when-clojure-spec`… :disappointed: